tsweb/tsweb.go

```diff
@@ -295,23 +309,54 @@ func (f ReturnHandlerFunc) ServeHTTPReturn(w http.ResponseWriter, r *http.Reques
 // Handled requests are logged using opts.Logf, as are any errors.
 // Errors are handled as specified by the Handler interface.
 func StdHandler(h ReturnHandler, opts HandlerOptions) http.Handler {
-	if opts.Now == nil {
-		opts.Now = time.Now
-	}
 	if opts.Logf == nil {
-		opts.Logf = logger.Discard
+		return Handler(h, opts)
 	}
-	return retHandler{h, opts}
+	return LogHandler(Handler(h, opts), opts)
 }
 
-// retHandler is an http.Handler that wraps a Handler and handles errors.
-type retHandler struct {
-	rh   ReturnHandler
+// LogHandler returns an http.Handler that logs to opts.Logf.
+// It logs both successful and failing requests.
+// The log line includes the first error returned to [Handler] within.
+// The outer-most LogHandler(LogHandler(...)) does all of the logging.
+func LogHandler(h http.Handler, opts HandlerOptions) http.Handler {
+	setDefaultOpts(&opts)
+	return logHandler{h, opts}
+}
+
+// Handler converts a ReturnHandler into a standard http.Handler.
+// Errors are handled as specified by the ReturnHandler interface.
+func Handler(h ReturnHandler, opts HandlerOptions) http.Handler {
+	setDefaultOpts(&opts)
+	return errorHandler{h, opts}
+}
+
+// HandlerFunc converts a handler func into a standard http.Handler.
+// Errors are handled as specified by the ReturnHandler interface.
+func HandlerFunc(h func(w http.ResponseWriter, r *http.Request) error, opts HandlerOptions) http.Handler {
+	return Handler(ReturnHandlerFunc(h), opts)
+}
```

Comment:
Is this func paying for itself, utility vs API surface complexity-wise? The caller can do this one line too?

Also we already have `type ErrorHandlerFunc` which looks so close to this, even if it's totally different. (This is called HandlerFunc and for a func that return an error ...)

Reply:
I agree it was not pulling its weight and the similarity to the type name was confusing. I've since removed this `HandlerFunc`.

The other callbacks are `OnStart` with type `OnStartFunc` and `OnCompletion` with type `OnCompletionFunc`. Perhaps we should rename the `OnError` field's type from `ErrorHandlerFunc` to `OnErrorFunc`?